.github/workflows/staticcheck.yml

```diff
@@ -0,0 +1,28 @@
+name: "staticcheck"
+on: ["push", "pull_request"]
+
+# see https://staticcheck.io/docs/running-staticcheck/ci/github-actions/
+
+jobs:
+  ci:
+	name: "staticcheck"
+	strategy:
+	  fail-fast: false
+	  matrix:
+		os: ["ubuntu-latest", "macOS-latest"]
+		go: ["1.17.x"]
+	runs-on: ${{ matrix.os }}
+	steps:
+	- uses: actions/checkout@v1
+	  with:
+		fetch-depth: 1
+	- uses: WillAbides/setup-go-faster@v1.7.0
+	  with:
+		go-version: ${{ matrix.go }}
+	- run: "go test ./..."
+	- run: "go vet ./..."
+	- uses: dominikh/staticcheck-action@v1.2.0
```

Comment:
I would use https://golangci-lint.run/contributing/quick-start/ instead that has this linter and many others